### DISCUSSION TITLE
Rate Limiting APIs

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/runtime#72283] [Q&A]
I haven't had a chance to play with this yet, but are there equivalent extensions for HttpClient that will cause it to gracefully respond and handle rate limits?

It would be great if I could write my code to be as quick as possible and have the code automatically slow down to match rate limit headers returned by a server.

**Comment by gfoidl** [comment]
Take a look at https://github.com/App-vNext/Polly which has ratelimiting capabilities.
